Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -273,6 +273,44 @@ HRESULT AddGeometryToPathWithTransformation(const ID2D1Geometry* geometry, const
         SetLastTransform(transform);
         return S_OK;
     }
+
+    HRESULT WidenByStroking(
+        CGPathRef path, CGFloat lineWidth, CGLineCap lineCap, CGLineJoin lineJoin, CGFloat miterLimit, const CGAffineTransform* transform) {
+        RETURN_IF_FAILED(path->ClosePath());
+        ComPtr<ID2D1StrokeStyle> newStrokeStyle = NULL;
+
+        ComPtr<ID2D1Factory> factory;
+        pathGeometry->GetFactory(&factory);
+
+        ComPtr<ID2D1PathGeometry> newPath;
+        ComPtr<ID2D1GeometrySink> newBackingSink;
+        RETURN_IF_FAILED(factory->CreatePathGeometry(&newPath));
+        RETURN_IF_FAILED(newPath->Open(&newBackingSink));
+
+        RETURN_IF_FAILED(factory->CreateStrokeStyle(D2D1::StrokeStyleProperties((D2D1_CAP_STYLE)lineCap,
+                                                                                (D2D1_CAP_STYLE)lineCap,
+                                                                                (D2D1_CAP_STYLE)lineCap,
+                                                                                (D2D1_LINE_JOIN)lineJoin,
+                                                                                lineWidth,
+                                                                                D2D1_DASH_STYLE_SOLID,
+                                                                                0),
+                                                    nullptr,
+                                                    0,
+                                                    &newStrokeStyle));
+
+        D2D1_MATRIX_3X2_F d2dTransform = D2D1::IdentityMatrix();
+        if (transform) {
+            d2dTransform = __CGAffineTransformToD2D_F(*transform);
+        }
+        RETURN_IF_FAILED(path->GetPathGeometry()->Widen(lineWidth, newStrokeStyle.Get(), d2dTransform, newBackingSink.Get()));
+
+        newBackingSink->SetFillMode(D2D1_FILL_MODE_WINDING);
+
+        pathGeometry = newPath;
+        geometrySink = Make<_CGPathCustomSink>(newBackingSink.Get());
```

Comment:
That's a good point, we need additional tests to track behavior beyond the creation of this widened/stroked path.